devnet-sdk/system/system.go

```diff
@@ -84,3 +96,17 @@ var _ InteropSystem = (*interopSystem)(nil)
 func (i *interopSystem) InteropSet() InteropSet {
 	return i.system // TODO: the interop set might not contain all L2s
 }
+
+func (i *interopSystem) Supervisor(ctx context.Context) (Supervisor, error) {
+	if i.supervisor != nil {
+		return i.supervisor, nil
+	}
+
+	cl, err := client.NewRPC(ctx, nil, i.supervisorRPC)
+	if err != nil {
+		return nil, fmt.Errorf("failed to dial supervisor RPC: %w", err)
+	}
+	supervisor := sources.NewSupervisorClient(cl)
+	i.supervisor = supervisor
+	return supervisor, nil
+}
```

Comment:
Are we doing lazy initialization as a rule or is it particularly useful in this context?

Reply:
that's a good question. My reasoning was that I really didn't want to fail the creation of the InteropSystem altogether if the supervisor endpoint is broken. So this is the next available spot where to create the client.

Besides, I didn't want to dial RPC clients for features that we might not use. That would be kinda solved with the WithLazyDial option just as well, but then we're in this weird situation where getting the interface cannot actually fail, and will get communication errors later.

All in all I felt it was the cleanest available option, but definitely willing to reconsider.